src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp

```diff
@@ -965,6 +983,182 @@ void HydrofractureSolver< POROMECHANICS_SOLVER >::setUpDflux_dApertureMatrix( Do
+        apertureF[newElemIndex] = aperture[newElemIndex];
+  #endif
+        if( m_newFractureInitializationType == InitializationType::Displacement )
+        {
+          aperture[newElemIndex] = 1.0e99;
+        }
+        arraySlice1d< localIndex const > const faceToEdges = facesToEdges[newElemIndex];
+
+        for( localIndex ke=0; ke<faceToEdges.size(); ++ke )
+        {
+          localIndex const edgeIndex = faceToEdges[ke];
+
+          auto connIter = edgesToConnectorEdges.find( edgeIndex );
+          if( connIter == edgesToConnectorEdges.end() )
+          {
+            return;
+          }
+          localIndex const connectorIndex = edgesToConnectorEdges.at( edgeIndex );
+          localIndex const numElems = fractureConnectorsToFaceElements.sizeOfArray( connectorIndex );
+
+          for( localIndex kfe=0; kfe<numElems; ++kfe )
+          {
+            localIndex const fractureElementIndex = fractureConnectorsToFaceElements[connectorIndex][kfe];
+
+            if( newFractureElements.count( fractureElementIndex ) == 0 )
+            {
+              initialPressure = std::min( initialPressure, fluidPressure_n[fractureElementIndex] );
+              initialAperture = std::min( initialAperture, aperture[fractureElementIndex] );
+            }
+          }
+        }
+        if( m_newFractureInitializationType == InitializationType::Pressure )
+        {
+          fluidPressure[newElemIndex] = initialPressure > 1.0e98? 0.0:initialPressure;
+          fluidPressure_n[newElemIndex] = fluidPressure[newElemIndex];
+        }
+        else if( m_newFractureInitializationType == InitializationType::Displacement )
+        {
+          // Set the aperture/fluid pressure for the new face element to be the minimum
+          // of the existing value, smallest aperture/pressure from a connected face element.
+          // aperture[newElemIndex] = std::min(aperture[newElemIndex], initialAperture);
+
+          localIndex const faceIndex0 = faceMap[newElemIndex][0];
+          localIndex const faceIndex1 = faceMap[newElemIndex][1];
+
+          localIndex const numNodesPerFace = faceToNodesMap.sizeOfArray( faceIndex0 );
+
+          bool zeroDisp = true;
+
+          for( localIndex a=0; a<numNodesPerFace; ++a )
+          {
+            localIndex const node0 = faceToNodesMap( faceIndex0, a );
+            localIndex const node1 = faceToNodesMap( faceIndex1, a==0 ? a : numNodesPerFace-a );
+            if( LvArray::math::abs( LvArray::tensorOps::l2Norm< 3 >( totalDisplacement[node0] ) ) > 1.0e-99 &&
+                LvArray::math::abs( LvArray::tensorOps::l2Norm< 3 >( totalDisplacement[node1] ) ) > 1.0e-99 )
+            {
+              zeroDisp = false;
+            }
+          }
+          if( zeroDisp )
+          {
+            aperture[newElemIndex] = 0;
+          }
+        }
+
+        if( this->getLogLevel() > 1 )
+        {
+          printf( "New elem index, init aper, init press = %4d, %4.2e, %4.2e \n",
```

Comment:
GEOS_LOG?

Reply:
honestly we should never output stuff inside a kernel but this kernel has to be on host anyways so I think it's fine.